### PR TITLE
retawq: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
the following recipe adds support for [retawq](https://retawq.sourceforge.net/)

```
> uname -a
Haiku shredder 1 hrev57104 Jun 22 2023 06:25:10 x86_64 x86_64 Haiku

> retawq --version
retawq 0.2.6c - Copyright (C) 2001-2006 Arne Thomassen <arne@arne-thomassen.de>
retawq is basically released under certain versions of the
GNU General Public License and WITHOUT ANY WARRANTY.
The project home page is <http://retawq.sourceforge.net/>.
```

### CODE REVIEW

File: www-client/retawq/retawq-0.2.6c.recipe [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libssl$secondaryArchSuffix
+	devel:libncurses${secondaryArchSuffix}
```
Comment (korli) [inline www-client/retawq/retawq-0.2.6c.recipe:28]:
```suggestion
	devel:libncurses$secondaryArchSuffix
	devel:libssl$secondaryArchSuffix
```

File: www-client/retawq/retawq-0.2.6c.recipe [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retawq$secondaryArchSuffix = $portVersion
+	cmd:retawq = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
```
Comment (korli) [inline www-client/retawq/retawq-0.2.6c.recipe:22]:
```suggestion
	lib:libncurses$secondaryArchSuffix
	lib:libssl$secondaryArchSuffix
```

File: www-client/retawq/retawq-0.2.6c.recipe [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libssl$secondaryArchSuffix
+	devel:libncurses${secondaryArchSuffix}
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config
```
Comment (korli) [inline www-client/retawq/retawq-0.2.6c.recipe:33]:
```suggestion
	cmd:pkg_config$secondaryArchSuffix
```

File: www-client/retawq/retawq-0.2.6c.recipe [hunk 4]
```diff
@@ -0,0 +1,46 @@
+SUMMARY="A multi-threaded text-based web browser"
+DESCRIPTION="an interactive, multi-threaded network client (web browser) for text \
```
Comment (korli) [inline www-client/retawq/retawq-0.2.6c.recipe:2]:
```suggestion
DESCRIPTION="An interactive, multi-threaded network client (web browser) for text \
```

File: www-client/retawq/retawq-0.2.6c.recipe [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config
+	"
+
+BUILD()
+{
+	LDFLAGS="-lnetwork " ./configure --set-tls=2
```
Comment (korli) [inline www-client/retawq/retawq-0.2.6c.recipe:38]:
```suggestion
	LDFLAGS="-lnetwork" ./configure --set-tls=2
```

Comment (Begasus) [reply www-client/retawq/retawq-0.2.6c.recipe:38]: needs `-lssl` in there also

**Comment by Begasus** [comment]
Build failure on 32bit:
```
gcc -O2  -D_REENTRANT -D_THREAD_SAFE   -lnetwork -lssl  init.o main.o parser.o resource.o stuff.o   -lncurses -o retawq
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `tls_initialize':
resource.c:(.text+0x14f5): undefined reference to `RAND_status'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x150f): undefined reference to `RAND_file_name'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x152d): undefined reference to `RAND_bytes'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1547): undefined reference to `RAND_write_file'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1640): undefined reference to `RAND_seed'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1645): undefined reference to `RAND_status'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x168d): undefined reference to `RAND_seed'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1692): undefined reference to `RAND_status'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x16bc): undefined reference to `RAND_load_file'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x16c1): undefined reference to `RAND_status'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1700): undefined reference to `RAND_seed'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.c:(.text+0x1705): undefined reference to `RAND_status'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `tls_openssl_shutdown':
resource.c:(.text+0x399a): undefined reference to `ERR_clear_error'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `__conn_read':
resource.c:(.text+0x3c4c): undefined reference to `ERR_clear_error'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `tls_do_handshaking':
resource.c:(.text+0x461a): undefined reference to `ERR_clear_error'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `conn_write_writedata':
resource.c:(.text+0x4825): undefined reference to `ERR_clear_error'
/boot/system/develop/tools/x86/bin/../lib/gcc/i586-pc-haiku/11.2.0/../../../../i586-pc-haiku/bin/ld: resource.o: in function `fetch_about':
resource.c:(.text+0xd16d): undefined reference to `OpenSSL_version'
```

File: www-client/retawq/retawq-0.2.6c.recipe (hunk 1 above)
Comment (hoanga) [reply www-client/retawq/retawq-0.2.6c.recipe]: updated

File: www-client/retawq/retawq-0.2.6c.recipe (hunk 2 above)
Comment (hoanga) [reply www-client/retawq/retawq-0.2.6c.recipe]: updated

File: www-client/retawq/retawq-0.2.6c.recipe (hunk 3 above)
Comment (hoanga) [reply www-client/retawq/retawq-0.2.6c.recipe]: updated

File: www-client/retawq/retawq-0.2.6c.recipe (hunk 4 above)
Comment (hoanga) [reply www-client/retawq/retawq-0.2.6c.recipe]: updated

File: www-client/retawq/retawq-0.2.6c.recipe (hunk 5 above)
Comment (hoanga) [reply www-client/retawq/retawq-0.2.6c.recipe]: updated and included `-lssl`

**Comment by hoanga** [comment]
thanks for the suggestions! have incorporated and pushed updates.

**Pull request merged**